src/index.js

```diff
@@ -327,22 +329,38 @@ class HCaptcha extends React.Component {
     }
 
     execute (opts = null) {
+
+      opts = typeof opts === 'object' && opts !== null ? opts : null;
+
       try {
         const { captchaId } = this.state;
         const hcaptcha = this._hcaptcha;
-
-
+        
         if (!this.isReady()) {
-            return;
-        }
-
-        if (opts && typeof opts !== "object") {
-            opts = null;
+          const onReady = new Promise((resolve, reject) => {
+            this._onReady = (id) => {
+              try {
+                if (opts && opts.async) {
+                  hcaptcha.execute(id, opts).then(resolve).catch(reject);
```

Comment:
it looks like `const hcaptcha = this._hcaptcha;` needs to be also added inside of the this._onReady function
after making this change, it seems to start working:
```
this._onReady = (id) => {
      try {
        const hcaptcha = this._hcaptcha;

        if (opts && opts.async) {
```

otherwise, I receive this when executing hCaptcha on the component mount:

<img width="1437" alt="Screenshot 2024-08-14 at 10 43 37" src="https://github.com/user-attachments/assets/3e4ad006-8be7-4dfb-8200-272af3eb754e">